src/controller/controller.cpp

```diff
@@ -207,11 +212,33 @@ void Controller::computeControl(double dt)
     xc_.r = saturate(PID_psi_.computePID(xc_.psi, xhat_.psi, dt), max_.yaw_rate, -max_.yaw_rate);
 
     // Rotate into body frame
-    /// TODO: Include pitch and roll in this mapping
-    xc_.u = saturate(pndot_c*cos(xhat_.psi) + pedot_c*sin(xhat_.psi), max_.u, -1.0*max_.u);
-    xc_.v = saturate(-pndot_c*sin(xhat_.psi) + pedot_c*cos(xhat_.psi), max_.v, -1.0*max_.v);
+    /// TODO: Include pitch and roll in this mapping  -- This is done now?
+    double ctheta = cos(xhat_.theta);
+    double stheta = sin(xhat_.theta);
+    double cphi = cos(xhat_.phi);
+    double sphi = sin(xhat_.phi);
+    double cpsi = cos(xhat_.psi);
+    double spsi = sin(xhat_.psi);
+
+    // Rotate velocities from vehicle to body frame
+    double pxdot = pndot_c*(ctheta*cpsi) + pedot_c*(ctheta*spsi) - pddot*(stheta);
+    double pydot = pndot_c*(sphi*stheta*cpsi - cphi*spsi) + pedot_c*(sphi*stheta*spsi + cphi*cpsi) + pddot*(sphi*ctheta);
+//    double pzdot = pndot_c*(cphi*stheta*cpsi + sphi*spsi) + pedot_c*(cphi*stheta*spsi - sphi*cpsi) + pddot_c*(cphi*ctheta);
+
```

Comment:
Could we use standard rotation methods here instead of using euler angles?  Somehow I feel like a quaternion multiply would make this a ton cleaner.